app/blueprints/api.py

```diff
@@ -0,0 +1,365 @@
+""" Blueprint for handling user requests against the database """
+
+from datetime import datetime
+import time
+import sqlite3
+import os.path
+
+from flask import Blueprint, g, request, current_app
+
+api = Blueprint(
+    "api",
+    __name__,
+)
+
+
+def get_db():
+    """
+    Establishes a connection to the database. Requires an app context to run.
+    See https://flask.palletsprojects.com/en/stable/patterns/sqlite3/ for more details.
+    """
+    db = getattr(g, "_database", None)
+    if db is None:
+        if os.path.isfile(current_app.config["DATABASE_PATH"]):
+            print(current_app.config["DATABASE_PATH"])
+            # If the database file exists, load it. Otherwise, init the database.
+            db = g._database = sqlite3.connect(current_app.config["DATABASE_PATH"])
+        else:
+            # Regenerate database
+            db = g._database = sqlite3.connect(current_app.config["DATABASE_PATH"])
+            with current_app.open_resource("db/schema.sql") as f:
+                db.cursor().executescript(f.read().decode("utf-8"))
+
+            # If we are testing load test data
+            if current_app.config["TESTING"]:
+                with current_app.open_resource("db/load_testing_data.sql") as f:
+                    db.cursor().executescript(f.read().decode("utf-8"))
+            db.commit()
+
+    db.row_factory = sqlite3.Row
+    return db
+
+
+def query_db(statement, args=(), one=False):
+    """Executes a SQL statement against the database and returns the results."""
+    cur = get_db().execute(statement, args)
+    results = cur.fetchall()
+    cur.close()
+
+    return (results[0] if results else None) if one else results
+
+
+@api.route("/student_start", methods=["POST"])
+def api_post_student_start():
+    """
+    Create a new student_submission object to show that a student began working on an exercise
+    Finds or creates new rows in the exercise, section, and student tables.
+    """
+    print("---POST START---")
+    # 1. Get time to log (do this first to maintain accuracy in timing!)
+    time_started = time.mktime(datetime.now().timetuple())
+
+    try:
+        print("Find/Create section")
+        # 1. Find or Create Section
+        # 1A. Query for section
+        section_query = query_db(
+            """
+            SELECT pk_section_id 
+            FROM section 
+            WHERE txt_section_name = ? 
+                AND txt_instructor_name = ?;
+            """,
+            (request.json["section_name"], request.json["instructor_name"]),
```

Comment:
Using name parameters and the `request.json` dictionary here should simplify most of these requests.

Reply:
This has been addressed in the latest push